Adding command line options to -ICP to skip translations

HeadLessHUN opened the pull request:
Add suboptions for icp to skip each translation.

-SKIP_TX
-SKIP_TY
-SKIP_TZ

#1898

dgirardeau commented on qCC/ccCommandLineCommands.cpp:
```diff
@@ -6292,23 +6295,27 @@ bool CommandICP::process(ccCommandLineInterface& cmd)
 				else if (rotation == "X")
 				{
-					transformationFilters = CCCoreLib::RegistrationTools::SKIP_RYZ;
+					transformationFilters |= CCCoreLib::RegistrationTools::SKIP_RYZ;
```
why the "|' operator? Did you imagine a use case where -ROT would be used multiple times for a single ICP command? It won't be considered by the registration algorithm anyway... The algorithm only considers one SKIP_R?? option at a time.

HeadLessHUN replied:
you are right, but what if someone send a -SKIP_TX before -ROT?

dgirardeau replied:
Ah ok, that's for the SKIP_T?, my bad...

dgirardeau commented on qCC/ccCommandLineCommands.cpp:
```diff
@@ -6292,23 +6295,27 @@ bool CommandICP::process(ccCommandLineInterface& cmd)
 				{
-					transformationFilters = CCCoreLib::RegistrationTools::SKIP_RYZ;
+					transformationFilters |= CCCoreLib::RegistrationTools::SKIP_RYZ;
+					cmd.print(QObject::tr("[ICP] Skip RYZ, current filter: %1").arg(transformationFilters));
```
Is it a 'debug' log?

HeadLessHUN replied:
it was kind of, but it is not that useless :), There were no logs about confirmations of arguments.

dgirardeau replied:
Ah ok, it's just that the value of 'transformationFilters' might not be super meaningful to the users... (so maybe the second part could be 'printDebug')

dgirardeau commented:
Ok, I realized that the 'Debug' versions of the print/warning/error methods of the logger were not readily accessible via the command line parser. So I added them, and moved the logging of 'transformationFilters' in a single final call to printDebug.

I also made sure that explicitly calling '-ROT XYZ' would reset the rotations (just in case).

HeadLessHUN commented:
> Ok, I realized that the 'Debug' versions of the print/warning/error methods of the logger were not readily accessible via the command line parser. So I added them, and moved the logging of 'transformationFilters' in a single final call to printDebug.
> 
> I also made sure that explicitly calling '-ROT XYZ' would reset the rotations (just in case).

isn't it more resource if we call QObject::tr() without argument or just use the quoted string in this case? (for the future)

dgirardeau commented:
So 'tr' is to allow translators to translate the string. So it's all a question of whether this is user facing and someone might want to translate it one day. That something that should have done everywhere, but sadly it's missing in a lot of places...

HeadLessHUN commented:
I think I'm done with this as well.
